Lesson18/bilge/script.js

```diff
@@ -0,0 +1,216 @@
+const testArrayEx5 = [1, 2, 3, 4, 7, 10];
+
+function findNewArrayWithEvenNumbers (array) {  
+  let newArr2 = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] % 2 === 0) {
+      newArr2.push(array[i]);  
+    }    
+  } 
+  return newArr2;
+}
+
+console.log('Exercise 5: ');
+
+console.log(findNewArrayWithEvenNumbers(testArrayEx5));
+
+
+// 6. Write a function that takes an array and a value, and returns a new array with all occurrences of the value removed.
+// Input: ([1, 2, 3, 4, 2], 2)
+// Output: [1, 3, 4]
+const testArrayEx6 = [1, 2, 3, 4, 2];
+
+function getNewArray(array, removedValue) {
+  let resultArr = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] !== removedValue) {
+      resultArr.push(array[i]);
+    }
+  }
+  return resultArr;
+}
+
+console.log("Exercise 6: ");
+
+console.log(getNewArray(testArrayEx6, 2));
+
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = { name: "John", age: 30, city: "New York" };
+
+function findKeysAsAnArray (object) {
+  let text = [];
+  for (let obj in object) {
+    text.push(obj);
+  }
+  return text;
+}
+
+console.log("Exercise 7: ");
+
+console.log(findKeysAsAnArray(testObjEx7));
+
+// 8. Write a function that takes an object and returns the values as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["John", 30, "New York"]
+const testObjEx8 = { name: "John", age: 30, city: "New York" };
+
+function findValuesOfObject (object) {
+  let returnValue = [];
+  for (let keys in object) {
+    returnValue.push(object[keys]);
+  }
+  return returnValue;
+}
+
+console.log("Exercise 8: ");
+
+console.log(findValuesOfObject(testObjEx8));
+
+// 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
+// Input: ({name: "John", age: 30, city: "New York"}, "age")
+// Output: true
+// Input: ({name: "John", age: 30, city: "New York"}, "surname")
+// Output: false
+const testObjEx9 = { name: "John", age: 30, city: "New York" };
+
+function findIfKeyExists (object, key) {
+  if (object.hasOwnProperty(key) === true) {    //checks if that key exists in the object and returns 'true' if that's the case
+    return true;
+  } else {
+  return false;
+  }
+}
+
+console.log("Exercise 9: ");
+
+console.log(findIfKeyExists(testObjEx9, "age"));
+console.log(findIfKeyExists(testObjEx9, "surname"));
+
+// 10. Write a function that takes an object and a key-value pair, and adds the key-value pair to the object.
+// Input: ({name: "John", age: 30}, "city", "New York")
+// Output: {name: "John", age: 30, city: "New York"}
+const testObjEx10 = { name: "John", age: 30 };
+
+function addKeyValuePair (obj, key, value) {
+  if (obj.hasOwnProperty(key) === false) {    //checks if that key does not exist in the object, and adds the key-value pair to the object
```

Comment:
When working with boolean values, you don't need to use comparison in the if statement.
```suggestion
  if (!obj.hasOwnProperty(key)) {
```